React/Base/macOS/RCTUIKit.m

```diff
@@ -812,13 +814,19 @@ - (void)setImage:(UIImage *)image
   CALayer *layer = [self layer];
   
   if ([layer contents] != image || [layer backgroundColor] != nil) {
-    if (_tintColor != nil) {
-      image = [image copy];
-      [image lockFocus];
-      [_tintColor set];
-      NSRect imageRect = { NSZeroPoint, image.size };
-      NSRectFillUsingOperation(imageRect, NSCompositingOperationSourceIn);
-      [image unlockFocus];
+    if (_tintColor) {
+      if (!_tintingLayer) {
+        _tintingLayer = [CALayer new];
+        [_tintingLayer setFrame:self.bounds];
+        [_tintingLayer setAutoresizingMask:kCALayerWidthSizable | kCALayerHeightSizable];
+        [_tintingLayer setZPosition:1.0];
+        [_tintingLayer setCompositingFilter:[CIFilter filterWithName:@"CISourceInCompositing"]];
```

Comment:
> "In macOS, after creating a filter with this method you must call setDefaults or set parameters individually by calling setValue:forKey:. In iOS, the filter’s parameters are automatically set to default values."

https://developer.apple.com/documentation/coreimage/cifilter/1438255-filterwithname?changes=latest_major

Do we need to do something here?

Reply:
I added the setDefaults method call so that we're sure the filter is initialized correctly on macOS.